src/classLibrary/Generator.ts

```diff
@@ -0,0 +1,121 @@
+
+import deleteEmpty from "delete-empty";
+import { PackageResolver } from "./ParsingStrategies/PackageResolver";
+import { PluginHandler } from "./PluginHandler";
+
+export class Generator {
+    private readonly util: Utility;
+    private guidGenerator: IGuidGenerator;
+
+    constructor(util: Utility, guidGenerator: IGuidGenerator) {
+        this.util = util;
+        this.guidGenerator = guidGenerator;
+    }
+    
+    async GenerateVFS(cyanSafe: CyanSafe, globFactory: GlobFactory): Promise<VirtualFileSystemInstance[]> {
+    
+        //package resolver may need to think how to expand it to beyond npm modules
+        let strategies: IParsingStrategy[] = [
+            new GuidResolver(this.guidGenerator, this.util),
+            new IfElseResolver(this.util),
+            new InlineFlagResolver(this.util),
+            new VariableResolver(this.util),
+            new PackageResolver(this.util)
+        ];
+
+        let parser: Parser = new Parser(this.util, strategies, cyanSafe);
+
+        console.log(chalk.greenBright("Preparation done!"));
+        console.log(chalk.cyanBright("Performing variable and flag scans..."));
+
+        //create globs (empty content vfs)
+        let filesMetadata: IFileSystemInstanceMetadata[] = cyanSafe.globs.Map((g: Glob) => globFactory.GenerateFilesMetadata(g, g.root)).Flatten();
+
+        //remove package.lock
+        filesMetadata = filesMetadata.Where(f => !f.sourceAbsolutePath.includes("node_modules"))
+        .Where(f => f.sourceAbsolutePath.ReplaceAll("\\\\", "/").split("/").Last()! !== "package-lock.json")
+        .Where(f => f.sourceAbsolutePath.ReplaceAll("\\\\", "/").split("/").Last()! !== "yarn.lock");
+
+        //no ignore is put
+        let files = globFactory.CreateEmptyFiles(filesMetadata);
+
+        //count for paths
+        parser.CountFiles(files);
+
+        //parse to get path according to parsing strategy
+        parser.ParseFiles(files);
+
+        //Read file asynchronously into the VFS (put the content with the metadata into VFS)
+        files = await globFactory.ReadFiles(files);
+
+        //count for the remaining content
+        let isNoUnusedFlags: boolean = parser.CountOccurence(files);
+        
+        //if there are unused flags
+        if (!isNoUnusedFlags) {
+
+        }        
+
+        console.log(chalk.cyanBright("Parsing template content..."));
+
+        //Parse Templates
+        files = parser.ParseContent(files);
+
+        //Generate warning of possible possible unaccounted flags
+        parser.CountPossibleRemains(files);
+
+        return files;
+    }
+
+    async GenerateTemplate(cyanSafe: CyanSafe, templatePath: string, folderName: string) {
+        //dest file path
+        //cwd is where the user is calling from
+        let destPath: string = path.resolve(process.cwd(), folderName);
+
+        //Check if the target path is empty
+        if (fs.existsSync(destPath)) {
+            //ask if want to delete existing stuff
```

Comment:
Can just add a TODO here 😆 

Implementation of delete existing stuff can just follow the oldCyanPrint